lib/modules/auth/presenter/auth_signin_page.dart

```diff
@@ -31,16 +32,32 @@ class AuthSignInState extends State<AuthSignInPage> {
 
   double getButtonBoxSize(BuildContext context) {
     bool isKeyboardClosed =
-        MediaQuery.of(context).viewInsets.bottom == 0 ? true : false;
+    MediaQuery.of(context).viewInsets.bottom == 0 ? true : false;
     double buttonBoxSize =
-        Theme.of(context).textTheme.headlineMedium!.fontSize! * 1.1 + 100.0;
+        Theme.of(context).textTheme.headline6!.fontSize! * 1.1 + 100.0;
 
     if (isKeyboardClosed) {
       buttonBoxSize += 100.0;
     }
     return buttonBoxSize;
   }
 
+  Future<void> _signIn() async {
+    try {
+      UserCredential userCredential =
+      await FirebaseAuth.instance.signInWithEmailAndPassword(
+        email: _emailController.text,
+        password: _passwordController.text,
+      );
+
+      // Sign in successful, do something
+      print('Signed in: ${userCredential.user}');
+    } on FirebaseAuthException catch (e) {
+      // Sign in failed, display error message
+      print('Sign-in error: $e');
+    }
+  }
+
```

Comment:
Essa parte do fluxo de Sign In deveria ficar no arquivo do Firebase pra não misturar as responsbilidades. Aqui existem regras de negócio só do Firebase, o arquivo é um presenter, ou seja deve ser responsável principalmente por montar a parte visual da página que o usuário vai interagir e chamar as funções que tratam das regras de negócios em outras classes